lib/resty/redis.lua

```diff
@@ -82,11 +86,51 @@ function _M.connect(self, ...)
         return nil, "not initialized"
     end
 
+    local params = {...}
+
+    if not params or #params < 2 then
+        return nil, "invalid arguments"
+    end
+
     self._subscribed = false
 
-    return sock:connect(...)
-end
+    local host = params[1]
+    local connect_opts, opts, err
+    if startswith(host, "unix:") then
```

Comment:
we should check the return value here, the return value should be `1`.

Reply:
we can use string.find here directly, no need another function.

Reply:
removed local function